packages/grafana-data/src/field/fieldComparers.ts

```diff
@@ -79,16 +79,26 @@ const falsyComparer = (a: unknown, b: unknown): number => {
   return 0;
 };
 
-const timestampIndexComparer = (values: number[], reverse: boolean): IndexComparer => {
+const timestampIndexComparer = (values: number[], reverse: boolean, nanos?: number[]): IndexComparer => {
   let mult = reverse ? -1 : 1;
+
+  if (nanos !== undefined) {
+    return (a: number, b: number): number => mult * (values[a] - values[b] || nanos[a] - nanos[b]);
+  }
+
   return (a: number, b: number): number => mult * (values[a] - values[b]);
 };
 
-const timeIndexComparer = (values: unknown[], reverse: boolean): IndexComparer => {
+const timeIndexComparer = (values: unknown[], reverse: boolean, nanos?: number[]): IndexComparer => {
+  const mult = reverse ? -1 : 1;
   return (a: number, b: number): number => {
     const vA = values[a];
     const vB = values[b];
-    return reverse ? timeComparer(vB, vA) : timeComparer(vA, vB);
+
+    if (nanos !== undefined) {
```

Comment:
same, just return different fns instead of checking for nanos in the sorter.

Reply:
like how it is now, or do we want to call another function in `fieldIndexComparer`?